gcloud/storage/blob.py

```diff
@@ -19,9 +19,10 @@
 import os
 import time
 import datetime
-from StringIO import StringIO
-import urllib
+from io import BytesIO
 
+import six
+from six.moves.urllib.parse import quote  # pylint: disable=F0401
```

Comment:
See comment from before. I previously ran into the import complaining (can't be found staticly in `six.py`) when using `six.moves.http_client.HTTPException` in `datastore.connection.Connection._request` (before sharing the storage exception code).

I just gave up and did `import six` and then used the fully qualified name in the code.